www/nginx/src/opnsense/scripts/nginx/ngx_auth.php

```diff
@@ -45,15 +45,21 @@ function password_auth_test($username, $password, $auth_server)
 
 function password_auth($auth_server = 'Local Database')
 {
-    if (!isset($_SERVER['PHP_AUTH_PW']) || !isset($_SERVER['PHP_AUTH_PW'])) {
+    if (!isset($_SERVER['PHP_AUTH_PW']) || !isset($_SERVER['PHP_AUTH_USER'])) {
         return false;
     }
     return password_auth_test($_SERVER['PHP_AUTH_USER'], $_SERVER['PHP_AUTH_PW'], $auth_server);
 }
 
-if (password_auth()) {
+if (empty($_SERVER['AUTH_SERVER'])) {
+    $auth_server = 'Local Database';
+} else {
+    $auth_server = $_SERVER['AUTH_SERVER'];
+}
+
+if (password_auth($auth_server)) {
     header("HTTP/1.1 200 OK");
 } else {
     header("HTTP/1.1 401 Authorization Required");
     header('WWW-Authenticate: Basic realm="OPNsense Protected Area - Authentication Required"');
-}
+}
```

Comment:
Your editor shouldn’t trim trailing EoL on last line

Reply:
I restored it